src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -68,14 +67,15 @@ yamlToMeta :: PandocMonad m
            -> m Meta
 yamlToMeta opts bstr = do
   let parser = do
-        meta <- yamlBsToMeta parseBlocks bstr
+        meta <- yamlBsToMeta (asBlocks <$> parseBlocks) bstr
         return $ runF meta defaultParserState
   parsed <- readWithM parser def{ stateOptions = opts } ""
   case parsed of
     Right result -> return result
     Left e       -> throwError e
 
-
+asBlocks :: Functor f => f (B.Many Block) -> f MetaValue
+asBlocks p = MetaBlocks . B.toList <$> p
```

Comment:
`asMetaBlocks` might be a better name?

Reply:
Sidenote: this is the same as `fmap toMetaValue`, where the latter is from Text.Pandoc.Builder.

Reply:
@tarleb, good point.